### PR TITLE
Document the Template and Grouping in ColumnChooser

### PR DESCRIPTION
**Pull request by svdimitr**

### CODE REVIEW

File: components/grid/columns/menu.md [hunk 1]
```diff
@@ -165,6 +250,8 @@ To hide a column from the Column Chooser set the `VisibleInColumnChooser` proper
 
 * If the Grid has a [frozen]({%slug grid-columns-frozen%}) column (`Locked="true"`), that column cannot be unfrozen from the column menu.
 
+* If you are using the [Column Chooser Template]({%slug grid-templates-column-chooser%}) or you are grouping the columns into [sections](#sections), it is recommended to add the `Title` parameter to all Grid Columns.
+
 ## See Also
 
-  * [Live Demo: Visible Columns](https://demos.telerik.com/blazor-ui/grid/columns)
+  * [Live Demo: Grid Custom Column Menu](https://demos.telerik.com/blazor-ui/grid/custom-column-menu)
```
Comment (marin-bratanov) [inline components/grid/columns/menu.md:257]: Consider linking both column menu demos here

File: components/grid/templates/column-chooser.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+tags: telerik,blazor,grid,templates,column,menu
+published: True
+position: 40
+---
+
+# Column Chooser Template
+
+The `Column Chooser Template` lets you control the rendering of the [ColumnChooser]({%slug grid-column-menu%}#column-chooser). It exposes a `context` object that contains a List of all columns in the Grid. 
+
+>note When the Template is setup the list of columns, rendered by default, would not be present.
```
Comment (marin-bratanov) [inline components/grid/templates/column-chooser.md:15]: Does not have to be a note

File: components/grid/templates/column-chooser.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+````
+
+>caption The result from the code snippet above
+
+![templated column chooser example](images/templates-column-chooser-example.png)
+
+## See Also
+
+ * [Live Demo: Grid Templates](https://demos.telerik.com/blazor-ui/grid/templates)
```
Comment (marin-bratanov) [inline components/grid/templates/column-chooser.md:88]: Could link the custom column menu demo as well. That demo should also cross-link the docs.

**Review by marin-bratanov — APPROVED** [review]

File: components/grid/columns/menu.md (hunk 1 above)
Comment (svdimitr) [reply components/grid/columns/menu.md:257]: Great idea, I have no idea why I missed it

File: components/grid/templates/column-chooser.md (hunk 3 above)
Comment (svdimitr) [reply components/grid/templates/column-chooser.md:88]: Done

File: components/grid/templates/column-chooser.md (hunk 2 above)
Comment (svdimitr) [reply components/grid/templates/column-chooser.md]: I agree, I was wondering which was the better option.

**Pull request merged**